File: Go-Backend/internal/models/stock.go
package models

import (
	"time"
)

type StrategyData struct {
	Ticker        string  `json:"Ticker" bson:"ticker"`
	Score         float64 `json:"score" bson:"score"`
	Close         float64 `json:"close" bson:"close"`
	Change        float64 `json:"change" bson:"change"`
	PercentChange float64 `json:"pct_change" bson:"pct_change"`
	Name          string  `json:"name" bson:"name"`
}

type StrategyResult struct {
	Bullish []StrategyData `json:"bullish" bson:"bullish"`
	Bearish []StrategyData `json:"bearish" bson:"bearish"`
}

type StockData struct {
	MA50        StrategyResult `json:"ma50" bson:"ma50"`
	RSI         StrategyResult `json:"rsi" bson:"rsi"`
	Volume      StrategyResult `json:"volume" bson:"volume"`
	Final       StrategyResult `json:"combined" bson:"combined"`
	LastUpdated time.Time      `json:"last_updated" bson:"last_updated"`
}

File: Backend/models/Stock.js
const mongoose = require("../config/MongoConfig");
const dotenv = require("dotenv");
dotenv.config();

const stockEntrySchema = new mongoose.Schema({
  ticker: { type: String, required: true },
  score: { type: Number, required: true },
  price: { type: Number, required: true },
  change: { type: Number, required: true },
  pct_change: { type: Number, required: true },
  name: { type: String, required: true },
});

const stockDirectionSchema = new mongoose.Schema({
  bullish: [stockEntrySchema],
  bearish: [stockEntrySchema],
});

const stockDataSchema = new mongoose.Schema({
  ma50: stockDirectionSchema,
  rsi: stockDirectionSchema,
  volume: stockDirectionSchema,
  combined: stockDirectionSchema,
  last_updated: { type: Date, default: Date.now },
});

const stockModel = mongoose.model(
  "stock",
  stockDataSchema,
  process.env.MONGO_COLLECTION || "stock_data"
);

module.exports = stockModel;
